token/cli/src/main.rs

```diff
@@ -943,11 +953,34 @@ fn command_burn(
     })
 }
 
+
+fn check_if_amount_overflows_supply(  
+    config: &Config,
+    token: Pubkey,
+    ui_amount : u64,
+) -> bool {
+    let supply = config.rpc_client.get_token_supply(&token).unwrap();
```

Comment:
if we're offline, we can't do this check, so consider just returning `false` if `config.sign_only`